spec/lib/wakes/redirect_mapper_spec.rb

```diff
@@ -174,8 +174,61 @@
     it 'creates a new resource' do
       described_class.new '/source', '/target', 'New Label'
 
-      expect(Wakes::Location.find_by(:path => '/target').resource)
+      expect(Wakes::Location.find_by(:host => nil, :path => '/target').resource)
         .to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
     end
   end
+
+  describe 'handles non-default hosts' do
+    context 'source is default, target is default' do
+      it 'points the source location to the target resource' do
+        described_class.new '/source', '/target', 'New Label'
+
+        expect(Wakes::Location.find_by(:host => nil, :path => '/target').resource)
+          .to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
+      end
+    end
+
+    context 'source is default, target is non-default' do
+      it 'points the source location to the target resource' do
+        described_class.new '/source', '/target', 'New Label', nil, 'some.host'
+        expect(Wakes::Location.find_by(:host => nil, :path => '/source').resource)
+          .not_to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
+        expect(Wakes::Location.find_by(:host => 'some.host', :path => '/target').resource)
+          .not_to have_wakes_graph(:canonical_location => '/target', :legacy_locations => ['/source'])
+        expect(Wakes::Location.find_by(:host => nil, :path => '/source').resource)
+          .to have_wakes_graph(:canonical_location => 'some.host/target', :legacy_locations => ['/source'])
```

Comment:
aren't all three of these expectations working on the same resource? If you really wanted, you could do `expect(Wakes::Location.find_by(:host => nil, :path => '/source').resource_id).to eq(Wakes::Location.find_by(:host => 'some.host', :path => '/target').resource_id)

Once you've proven that, it's unnecessary to have both the `to` and the `not_to` expectations here. If it has one wakes graph, it necessarily doesn't have the other.